sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -479,14 +468,15 @@ case class DataSource(
       outputPath = outputPath,
       staticPartitions = Map.empty,
       ifPartitionNotExists = false,
-      partitionColumns = partitionAttributes,
+      partitionColumns = partitionColumns.map(UnresolvedAttribute.quoted),
       bucketSpec = bucketSpec,
       fileFormat = format,
       options = options,
       query = data,
```

Comment:
How about adding an `AnalysisBarrier` here?

Reply:
I think @cloud-fan wants `AnalysisBarrier` to be in trait `DataWritingCommand` to make sure the `query` is analyzed.
Adding `AnalysisBarrier` will not be helpful, the query should be analyzed here.